src/control-plane/backend/lambda/api/common/utils.ts

```diff
@@ -258,9 +258,47 @@ function getStackPrefix(): string {
   return `${prefix}-${SolutionInfo.SOLUTION_SHORT_NAME}`;
 }
 
-function getStackName(pipelineId: string, key: PipelineStackType, sinkType: string): string {
+function getIngestionStackTemplateUrl(state: WorkflowState | undefined, pipeline: IPipeline): string | undefined {
+  let templateUrl: string | undefined;
+  if (!state) {
+    return templateUrl;
+  }
+  if (state.Type === WorkflowStateType.PARALLEL) {
+    for (let branch of state.Branches as WorkflowParallelBranch[]) {
+      for (let key of Object.keys(branch.States)) {
+        templateUrl = getIngestionStackTemplateUrl(branch.States[key], pipeline);
+        if (templateUrl) {
+          return templateUrl;
+        }
+      }
+    }
+  } else if (state.Type === WorkflowStateType.STACK) {
+    if (state.Data?.Input.StackName.includes(PipelineStackType.INGESTION)) {
+      templateUrl = state.Data.Input.TemplateURL;
+    }
+  }
+  return templateUrl;
+}
+
+function getIngestionStackType(pipeline: IPipeline): PipelineStackType {
+  if (pipeline.workflow?.Workflow) {
+    const ingestionStackTemplateUrl = getIngestionStackTemplateUrl(pipeline.workflow?.Workflow, pipeline);
+    if (ingestionStackTemplateUrl && ingestionStackTemplateUrl?.includes('/ingestion-server-v2-stack.template.json')) {
```

Comment:
We need to make sure the existing pipeline with ingestion server v1 continue to using v1 template whatever the pipeline version is.

Do we have a test case to verfiy this behavior?

Reply:
Yes, we had the test case.